### PR TITLE
fix: fix searchPlan metricType modified concurrently

### PR DESCRIPTION
**Pull request by xige-16**
issue: #30225
/kind bug
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querynodev2/segments/plan.go [hunk 1]
```diff
@@ -42,19 +43,36 @@ type SearchPlan struct {
...
+		newPlan.delete()
+		return nil, common.InvalidFieldID, err
+	}
+
+	cMetricType := C.GetMetricTypeByFieldID(col.collectionPtr, fieldID)
+	defer C.free(unsafe.Pointer(cMetricType))
+	metricTypeInIndex := C.GoString(cMetricType)
+
+	if len(metricType) != 0 && metricType != metricTypeInIndex {
+		return nil, common.InvalidFieldID, merr.WrapErrParameterInvalid(metricTypeInIndex, metricType, "metric type not match")
```
Comment (czs007) [inline internal/querynodev2/segments/plan.go:71]: should release newPlan.

**Comment by xige-16** [comment]
rerun ut

**Pull request merged**